### PR TITLE
ci: Run code coverage for pull requests

### LINKED ISSUE #489: [infrastructure] Report codecover on PRs
We use codecov to report the code coverage on `main`, we should also activate the bot to inform us about potential dropping of coverage before merging a PR.

_Note_: In the intend to have separate sections (crates, integrations, etc) on this as a mono-repo, we probably want to report them as different 'flags' in the CI, allowing us to have different limits per area.

### PR DESCRIPTION
**Pull request by zecakeh**
Closes #489.

### CODE REVIEW

**Comment by poljar** [comment]
Do we want to add some settings to this? For example:

```yaml
coverage:
  status:
    project:
      default:
        # Commits pushed to master should not make the overall
        # project coverage decrease by more than 1%:
        target: auto
        threshold: 1%
    patch:
      default:
        # Be tolerant on slight code coverage diff on PRs to limit
        # noisy red coverage status on github PRs.
        # Note: The coverage stats are still uploaded
        # to codecov so that PR reviewers can see uncovered lines
        target: auto
        threshold: 1%
```

**Comment by zecakeh** [comment]
By the way it seems that v1 of the codecov action is deprecated. Should we change to v2 or v3?

**Comment by poljar** [comment]
> By the way it seems that v1 of the codecov action is deprecated. Should we change to v2 or v3?

Sure.

File: codecov.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        target: auto
+        threshold: 1%
+    patch:
+      default:
+        # Be tolerant on slight code coverage diff on PRs to limit
+        # noisy red coverage status on github PRs.
+        # Note: The coverage stats are still uploaded
+        # to codecov so that PR reviewers can see uncovered lines
+        target: auto
+        threshold: 1%
```
Comment (gnunicorn) [inline codecov.yaml:16]: newline

```suggestion
        threshold: 1%

```

**Review by gnunicorn — APPROVED** [review]
style only

**Pull request merged**